packages/core/src/indexing/local-index-api.ts

```diff
@@ -17,27 +19,43 @@ import { Model } from '@ceramicnetwork/stream-model'
 import { ISyncQueryApi } from '../sync/interfaces.js'
 
 /**
- * Takes a Model StreamID, loads it, and returns the IndexModelArgs necessary to prepare the
+ * Takes a ModelFieldsIndex, loads it, and returns the IndexModelArgs necessary to prepare the
  * database for indexing that model.
  */
 async function _getIndexModelArgs(
-  modelStreamId: StreamID,
-  repository: Repository
+  req: ModelFieldsIndex,
+  repository: Repository,
+  databaseIndexApi?: DatabaseIndexApi
 ): Promise<IndexModelArgs> {
+  const modelStreamId = req.streamID
   if (modelStreamId.type != Model.STREAM_TYPE_ID && !modelStreamId.equals(Model.MODEL)) {
     throw new Error(`Cannot index ${modelStreamId.toString()}, it is not a Model StreamID`)
   }
 
+  let opts: IndexModelArgs = {
+    model: modelStreamId,
+  }
+
   if (modelStreamId.type == Model.STREAM_TYPE_ID) {
     const modelState = await repository.load(modelStreamId, {})
     const content = modelState.state.next?.content ?? modelState.state.content
     Model.assertVersionValid(content, 'major')
     if (content.relations) {
-      return { model: modelStreamId, relations: content.relations }
+      opts = {
+        ...opts,
+        relations: content.relations,
+      }
+    }
+    const indices = (await databaseIndexApi?.getFieldsIndex(modelStreamId)) ?? req.indices
```

Comment:
what's happening here?  If the database already has indices for this model we take those instead of the ones provided with the command?  How could that happen?  Even if it did, would ignoring the indices provided by the command be the right move?  Seems like it would be surprising behavior if you specified some indices and those were not used at all

Reply:
ping